mailbox/client_conn.go

```diff
@@ -289,38 +283,28 @@ func (c *ClientConn) recv(ctx context.Context) ([]byte, error) {
 		default:
 		}
 
-		c.receiveMu.Lock()
-		_, msg, err := c.receiveSocket.Read(ctx)
+		var (
+			msg       []byte
+			retry     bool
+			errStatus ClientStatus
+			err       error
+		)
+		msg, retry, errStatus, err = c.transport.Recv(ctx)
```

Comment:
Hmm, why not use the `:=` syntax here instead of declaring all variables before?

Reply:
oh yeah hahaha sorry this was part of a refactor I was doing where I had some nested if-statements. Will fix!

Reply:
done!